src/main/kotlin/ru/hse/dynamomock/service/AWSDynamoDBMockService.kt

```diff
@@ -208,13 +215,96 @@ class AWSDynamoDBMockService(private val storage: DataStorageLayer) {
                 storage.updateItem(it)
             }
         }
-        deleteItemRequests.forEach{ storage.deleteItem(it) }
+        deleteItemRequests.forEach { storage.deleteItem(it) }
 
         return BatchWriteItemResponse.builder()
             .build()
     }
 
-    private fun getKeyFromMetadata(keyName: String, keys: Map<String, AttributeValue>, attributeDefinitions: List<AttributeDefinition>): Key {
+    fun scanItems(fileName: String, tableName: String) {
+        val allowedTypes = EnumSet.of(SS, S, N, NS, L, M, BOOL, NULL)
+        val columnsInfo = mutableListOf<Pair<String, String>>()
+
+        File(fileName).useLines { line ->
+            val firstRow = line.firstOrNull() ?: throw AWSMockCSVException("The file is empty")
+            val info = firstRow.split(";")
+            info.forEach {
+                val value = it.split("|")
+                if (value.size != 2) {
+                    throw AWSMockCSVException("Wrong value format. Use <column_name>|<type>")
+                }
+                val columnName = value[0].trim()
+                val type = value[1].trim()
+
+                if (!allowedTypes.contains(DynamoType.valueOf(type))) {
+                    throw AWSMockCSVException("Function scanItems supports only S, N, NS, SS, NULL, BOOL types right now")
+                }
+                columnsInfo.add(columnName to type)
+            }
+        }
+
+        var i = 0
+        val itemsList = mutableListOf<Map<String, AttributeValue>>()
+        File(fileName).forEachLine {
+            if (i != 0) {
+                val columnValues = it.split(";").map { v -> v.trim() }
+                if (columnValues.size != columnsInfo.size) {
+                    throw AWSMockCSVException("${i + 1} row's size doesn't match the size of first row")
+                }
+                val item = columnValues.mapIndexed { index, element ->
+                    val (columnName, type) = columnsInfo[index]
+                    columnName to when (DynamoType.valueOf(type)) {
```

Comment:
It seems you have become a huge fan of Pairs :) It's OK, especially after Java, like a breath of fresh air